src/Psalm/Internal/Analyzer/Statements/EchoAnalyzer.php

```diff
@@ -38,7 +38,7 @@ public static function analyze(
             $expr_type = $statements_analyzer->node_data->getType($expr);
 
             if ($statements_analyzer->data_flow_graph instanceof TaintFlowGraph) {
-                if ($expr_type) {
+                if ($expr_type && $expr_type->hasObjectType()) {
```

Comment:
Note: this was done to fix https://github.com/vimeo/psalm/issues/3709 originally but this had the side-effect of turning any int into a `numeric-string`. I was not comfortable skipping taint analysis for a possible string, so I reduced the scope of this fix in order to retrieve the real int type later.